packages/x-components/src/components/__tests__/dynamic-props.mixin.spec.ts

```diff
@@ -0,0 +1,41 @@
+import Vue from 'vue';
+import { mount, Wrapper } from '@vue/test-utils';
+import { ExtractArrayItems } from '@empathyco/x-utils';
+import { dynamicPropsMixin } from '../dynamic-props.mixin';
+
+const renderComponent = ({
+  props = ['list', 'button'] as const
+}: ComponentOptions = {}): ComponentAPI => {
+  const wrapper = mount({
+    mixins: [dynamicPropsMixin<ExtractArrayItems<typeof props>>(props)],
+    render: h => h(),
+    props: ['data']
+  });
+  return { wrapper };
+};
+
+describe('dynamicPropsMixin', () => {
+  it('expects to have the defined props from the mixin', () => {
+    const { wrapper } = renderComponent();
+    const props = Object.keys(wrapper.props());
+    expect(props).toHaveLength(3);
+    expect(props.includes('list')).toBeTruthy();
+    expect(props.includes('button')).toBeTruthy();
+    expect(props.includes('data')).toBeTruthy();
+  });
+});
+
+/**
+ * The options for the `renderComponent` function.
+ */
+interface ComponentOptions {
+  props?: readonly string[];
+}
```

Comment:
After talking with you I realised that complicating code here to have types makes no sense. In the end you have no types as the testing wrapper has the `Wrapper<Vue>` type. We could complicate it and return an `ExtendedVue` like you did for the mixin factory, but (for one time) I think it is complicating things unnecessarily. So you could simply remove the readonly stuff if you want and forget about literal types here.